prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -18,6 +19,7 @@ pub enum MirPureEncoderError {
     UnsupportedTerminator,
 }
 
+// TODO: does this need to be `&'vir [..]`?
```

Comment:
Is it the lifetime that's the problem? We could also add a `for` lifetime to the lazy methods, then use it as needed, e.g. here.

Reply:
I'd want a `&'a [vir::Expr<'vir>]` instead: the vector probably doesn't need to be interned. A `for` for the lazy methods doesn't work I'm pretty sure since you do need to get `&'vir VirCtxt` if you want to allocate stuff with a `'vir` lifetime

Reply:
I meant to have both the `'vir` lifetime *and* another `for` lifetime for calls.